src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -179,6 +179,25 @@ class ObservationStore(
               .where(PLANTINGS.PLANTING_SITE_ID.eq(OBSERVATIONS.PLANTING_SITE_ID))
               .and(PLANTINGS.PLANTING_SUBZONE_ID.isNotNull))
 
+  /**
+   * Returns a list of observations that are starting in 1 month or less and for which we have yet
+   * to send out notifications that they're coming up.
+   */
+  fun fetchNonNotifiedUpcomingObservations(): List<ExistingObservationModel> {
+    val maxStartDate =
+        LocalDate.ofInstant(clock.instant(), ZoneOffset.UTC).plusMonths(1).plusDays(1)
+
+    return fetchWithDateFilter(
+        listOf(
+            OBSERVATIONS.STATE_ID.eq(ObservationState.Upcoming),
+            OBSERVATIONS.START_DATE.le(maxStartDate),
+            OBSERVATIONS.UPCOMING_NOTIFICATION_SENT_TIME.isNull,
+            plantingSiteHasPlantings,
+        )) { todayAtSite, record ->
+          record[OBSERVATIONS.START_DATE]!! <= todayAtSite.plusMonths(1)
```

Comment:
Just for my understanding.. we wouldn't need to filter out observations with start dates that are back by more than a month - because they would be tagged as Overdue and not covered by this search, yes?